channels/views/frontpage.py

```diff
@@ -36,6 +37,7 @@ def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
         pagination, posts = get_pagination_and_posts(paginated_posts, listing_params)
         users = lookup_users_for_posts(posts)
         posts = [post for post in posts if post.author and post.author.name in users]
+        posts = proxy_posts(posts)
```

Comment:
Maybe combine with the above line, here and in similar code from `channels/views/posts.py`?
```python
posts = proxy_posts([post for post in posts if post.author and post.author.name in users])
```